File: Task_3/task_3.kt
import kotlin.math.abs

// Функция для вычисления суммы и произведения цифр числа
fun calculateSumAndProduct(n: Int): Pair<Int, Int> {
    var sum = 0
    var product = 1
    val number = n.toString() // Преобразуем число в строку

    for (digitChar in number) {
        val digit = digitChar.toString().toInt() // Преобразуем символ цифры в число
        sum += digit
        product *= digit
    }
    return Pair(sum, product)
}

fun main() {
    // Ввод чисел из стандартного ввода
    print("Введите числа через пробел: ")
    val input = readlnOrNull() ?: "" // Считываем строку из ввода
    val numbers = input.split(" ").mapNotNull { it.toIntOrNull() }  // Разделяем строку по пробелам и преобразуем в числа

    // Обработка каждого введенного числа
    for (n in numbers) {
        val (sum, product) = calculateSumAndProduct(n) // Получаем сумму и произведение
        val difference = abs(sum - product) // Вычисляем модуль разности

        // Формируем строку для вывода разложения числа
        val numberStr = n.toString()
        val sumExpression = numberStr.map { it.toString().toInt() }.joinToString(" + ")
        val productExpression = numberStr.map { it.toString().toInt() }.joinToString(" * ")

        println("$sum $product $difference ($sumExpression = $sum; $productExpression = $product; |$sum - $product| = $difference)")
    }
}

File: Task_3/task_3.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Program
{
    // Функция для вычисления суммы и произведения цифр числа
    static void CalculateSumAndProduct(int n, out int sum, out int product)
    {
        sum = 0;
        product = 1;
        string number = n.ToString(); // Преобразуем число в строку

        foreach (char digitChar in number)
        {
            int digit = digitChar - '0'; // Преобразуем символ цифры в число
            sum += digit;
            product *= digit;
        }
    }

    public static void Main(string[] args)
    {
        List<int> numbers = new List<int>();
        Console.WriteLine("Введите числа через пробел:");
        string input = Console.ReadLine();
        string[] numberStrings = input.Split(' '); // Разделяем ввод на отдельные строки

        foreach (string numberString in numberStrings)
        {
            if (int.TryParse(numberString, out int num)) // Проверяем, является ли строка числом
            {
                numbers.Add(num);
            }
        }

        // Обработка каждого введенного числа
        foreach (int n in numbers)
        {
            int sum, product;
            CalculateSumAndProduct(n, out sum, out product); // Вызываем функцию для вычисления суммы и произведения
            int difference = Math.Abs(sum - product); // Вычисляем модуль разности

            // Формируем вывод разложения
            string numberStr = n.ToString();
            string sumExpression = string.Join(" + ", numberStr.Select(c => c - '0'));
            string productExpression = string.Join(" * ", numberStr.Select(c => c - '0'));
            Console.WriteLine($"{sum} {product} {difference} ({sumExpression} = {sum}; {productExpression} = {product}; |{sum} - {product}| = {difference})");
        }
    }
}
